### DISCUSSION TITLE
Taskbar, Tray, Explorer and the Start menu are always not updating their theme properly.

### DISCUSSION THREAD

**Discussion by PeopleInside** [AutoDarkMode/Windows-Auto-Night-Mode#627] [Q&A]
Hi, since some mounts when Auto-Dark mode switch the theme from Windows light to Windows dark or viceversa the start menu is not updated. I need all time click on the start button to have it updated.

Any way to fix this issue?

- Commit: `212dbb4`
- Service/App: `10.3.0.90`
- Updater: `2.5.2`
- Shell: `1.3.1.0`
- .Net: `6.0.10`
- Windows: `22621.819`

**Comment by Spiritreader** [comment]
> Any way to fix this issue?

No, this is a known Windows bug.
You can find more information about what works and what doesn't here:
https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/wiki/Known-limitations

You can however tell us precisely what steps are necessary to reproduce this issue and attach your ADM config and log file.
It's also highly recommended that you update to the latest beta build (10.3.1.11) before doing so.

If you say you get this 100% of the time then your case is especially interesting, because that would mean either something is especially wrong with your Windows installation, or the mitigation used by ADM isn't applying properly.

For example, on my 3 test machines the failure rate is ~1 in 1000 switches for the taskbar and ~1 in 100 for explorer.


Not all users experience this issue the same way. We have found a few mitigations to reduce half-baked theme switches, but there will always remain some edge cases until MS decides to fix it.

**Comment by PeopleInside** [comment]
@Spiritreader thank you for your reply and suggestions.
I have only one PC and this issue happen always since some months.
I set Auto Dark mode to switch the theme to change desktop and the theme when night time arrive.

Desktop and Windows are updated but the start menu not and I have always to click on the start menu.

Steps are simply to set auto dark mode to switch themes

![immagine](https://user-images.githubusercontent.com/5006150/202854078-93f55c6e-91eb-4d4b-ad86-0d3b3ea3390a.png)

at custom time

![immagine](https://user-images.githubusercontent.com/5006150/202854120-6b2e60e0-b226-42d9-9b11-1ea27f76df2d.png)

then when PC reach this time the start menu are not updated until you don't click the start menu

Config.yml

---

```
AutoThemeSwitchingEnabled: true
Governor: Default
Autostart:
  Validate: true
WindowsThemeMode:
  Enabled: true
  DarkThemePath: C:\Windows\Resources\Themes\dark.theme
  LightThemePath: C:\Windows\Resources\Themes\aero.theme
  MonitorActiveTheme: true
  ApplyFlags: []
AppsSwitch:
  Enabled: true
  Component:
    Mode: Switch
SystemSwitch:
  Enabled: true
  Component:
    Mode: Switch
    TaskbarSwitchDelay: 1200
    TaskbarColorOnAdaptive: false
    TaskbarColorWhenNonAdaptive: Light
    DWMPrevalenceSwitch: false
    DWMPrevalenceEnableTheme: Light
ColorFilterSwitch:
  Enabled: false
  Component: {}
OfficeSwitch:
  Enabled: false
  Component:
    Mode: Switch
    LightTheme: 0
    DarkTheme: 4
Sunrise: 2022-10-04T07:45:00.0000000
Sunset: 2022-10-04T18:15:00.0000000
Location:
  PollingCooldownTimeSpan: 1.00:00:00
  Enabled: false
  UseGeolocatorService: true
  SunsetOffsetMin: 0
  SunriseOffsetMin: 0
  CustomLat: 0
  CustomLon: 0
Tunable:
  BatterySliderDefaultValue: 25
  DisableEnergySaverOnThemeSwitch: false
  UseLogonTask: false
  Debug: false
  Trace: false
  DebugTimerMessage: false
  ShowTrayIcon: true
  UICulture: it
GPUMonitoring:
  Enabled: false
  Threshold: 30
  MonitorTimeSpanMin: 1
  Samples: 1
Events:
  DarkThemeOnBattery: false
  Win10AllowLockscreenSwitch: true
Notifications:
  OnAutoThemeSwitching: true
  OnSkipNextSwitch: true
AutoSwitchNotify:
  Enabled: false
  GracePeriodMinutes: 5
Hotkeys:
  Enabled: true
  ForceLight: 
  ForceDark: 
  NoForce: 
  ToggleTheme: 
  TogglePostpone: 
  ToggleAutoThemeSwitch: 
IdleChecker:
  Enabled: false
  Threshold: 5
WallpaperSwitch:
  Enabled: false
  Component:
    TypeLight: Individual
    TypeDark: Individual
    Position: Fill
    GlobalWallpaper:
      Light: 
      Dark: 
    SolidColors:
      Light: '#FFFFFF'
      Dark: '#000000'
    Monitors: []
Updater:
  AutoInstall: true
  Enabled: true
  Silent: false
  DaysBetweenUpdateCheck: 1
  CheckOnStart: true
  VersionQueryUrl: 
  DownloadBaseUrl: 
  ZipCustomUrl: 
  HashCustomUrl: 
```

---

**Comment by Spiritreader** [comment]
Thanks for the log files.
However, unfortunately with the latest beta I am unable to reproduce this issue entirely.
Even when spamming theme switches faster than windows can process, it still does not exhibit your described behavior.

There are some outstanding issues with wallpaper parsing for default themes in 10.3.0.90, so maybe that is it.

I once again strongly advise to update to latest beta versions.

https://user-images.githubusercontent.com/18438899/202860990-d1448938-fb64-45fc-b122-0caee597f2cd.mp4

**Comment by PeopleInside** [comment]
When the actual beta will beta will be stable?
Yes the beta work I should keep maybe that and wait to switch to stable? Thanks

**Reply by Spiritreader** [reply]
> When the actual beta will beta will be stable?

We usually don't provide ETAs.

> Yes the beta work I should keep maybe that and wait to switch to stable? Thanks

In general, we advise people to only use stable if the beta isn't working for them. 
Stable is a branch where we push changes conservatively and after a testing period. 
Which means things won't break as often, but new fixes and features won't show up as soon as they're ready.

**Comment by PeopleInside** [comment]
The issue still be present, also in the beta if the PC has dark theme at night and get turned off then in the morning, when I turn on, Auto Dark Mode change the theme to light but the start menu will stay dark with text not readable.

If I switch the Windows 11 theme manually from dark to light and viceversa no issue.
Why the issue exist only if Auto Dark mode set the theme in the morning from dark to light?

**Reply by PeopleInside** [reply]
Created an issue https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/issues/674
